synapse/events/validator.py

```diff
@@ -12,10 +12,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import collections.abc
-from typing import List, Type, Union, cast
+from typing import TYPE_CHECKING, List, Type, Union, cast
 
 import jsonschema
-from pydantic import Field, StrictBool, StrictStr
+
+if TYPE_CHECKING:
+    from pydantic.v1 import Field, StrictBool, StrictStr
+else:
+    try:
+        from pydantic.v1 import Field, StrictBool, StrictStr
+    except ImportError:
+        from pydantic import Field, StrictBool, StrictStr
```

Comment:
Should this one be the same pattern of `if TYPE_CHECKING or HAS_PYDANTIC_V2`?

Reply:
Yes, it should. I'll fix that.

Reply:
I fixed the import.